In [1]:
# import libraries 
import pandas as pd
from pathlib import Path
import matplotlib.pyplot as plt

## Data

This dataset can be dowloaded from : https://data.world/raghav333/fifa-players

In [2]:
#read in the CSV as a dataframe 
df = pd.read_csv('Resources/fifa_cleaned.csv')
df.head()

id          name                       full_name  birth_date  age  \
0  158023      L. Messi  Lionel Andrés Messi Cuccittini  1987-06-24   31   
1  190460    C. Eriksen    Christian  Dannemann Eriksen  1992-02-14   27   
2  195864      P. Pogba                      Paul Pogba  1993-03-15   25   
3  198219    L. Insigne                 Lorenzo Insigne  1991-06-04   27   
4  201024  K. Koulibaly               Kalidou Koulibaly  1991-06-20   27   

   height_cm  weight_kgs  positions nationality  overall_rating  ...   lwb  \
0     170.18        72.1   CF,RW,ST   Argentina              94  ...  64+2   
1     154.94        76.2  CAM,RM,CM     Denmark              88  ...  71+3   
2     190.50        83.9     CM,CAM      France              88  ...  76+3   
3     162.56        59.0      LW,ST       Italy              88  ...  63+3   
4     187.96        88.9         CB     Senegal              88  ...  73+3   

    ldm   cdm   rdm   rwb    lb   lcb    cb   rcb    rb  
0  61+2  61+2  61+2  64+2  59+2  48+2  48+2  48+2  59+2  
1  71+3  71+3  71+3  71+3  66+3  57+3  57+3  57+3  66+3  
2  77+3  77+3  77+3  76+3  74+3  72+3  72+3  72+3  74+3  
3  58+3  58+3  58+3  63+3  58+3  44+3  44+3  44+3  58+3  
4  77+3  77+3  77+3  73+3  76+3  85+3  85+3  85+3  76+3  

[5 rows x 92 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17954 entries, 0 to 17953
Data columns (total 92 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   id                            17954 non-null  int64  
 1   name                          17954 non-null  object 
 2   full_name                     17954 non-null  object 
 3   birth_date                    17954 non-null  object 
 4   age                           17954 non-null  int64  
 5   height_cm                     17954 non-null  float64
 6   weight_kgs                    17954 non-null  float64
 7   positions                     17954 non-null  object 
 8   nationality                   17954 non-null  object 
 9   overall_rating                17954 non-null  int64  
 10  potential                     17954 non-null  int64  
 11  value_euro                    17699 non-null  float64
 12  wage_euro                     17708 non-null  float64
 13  p

In [4]:
df.describe()

id           age     height_cm    weight_kgs  \
count   17954.000000  17954.000000  17954.000000  17954.000000   
mean   215411.087780     25.565445    174.946921     75.301047   
std     29758.387106      4.705708     14.029449      7.083684   
min        16.000000     17.000000    152.400000     49.900000   
25%    201117.250000     22.000000    154.940000     69.900000   
50%    222919.000000     25.000000    175.260000     74.800000   
75%    237613.500000     29.000000    185.420000     79.800000   
max    247607.000000     46.000000    205.740000    110.200000   

       overall_rating     potential    value_euro      wage_euro  \
count    17954.000000  17954.000000  1.769900e+04   17708.000000   
mean        66.240169     71.430935  2.479280e+06    9902.134628   
std          6.963730      6.131339  5.687014e+06   21995.593750   
min         47.000000     48.000000  1.000000e+04    1000.000000   
25%         62.000000     67.000000  3.250000e+05    1000.000000   
50%         66.000000     71.000000  7.000000e+05    3000.000000   
75%         71.000000     75.000000  2.100000e+06    9000.000000   
max         94.000000     95.000000  1.105000e+08  565000.000000   

       international_reputation_1_5  weak_foot_1_5  ...     penalties  \
count                  17954.000000   17954.000000  ...  17954.000000   
mean                       1.111674       2.945695  ...     48.357302   
std                        0.392168       0.663691  ...     15.810844   
min                        1.000000       1.000000  ...      5.000000   
25%                        1.000000       3.000000  ...     38.000000   
50%                        1.000000       3.000000  ...     49.000000   
75%                        1.000000       3.000000  ...     60.000000   
max                        5.000000       5.000000  ...     92.000000   

          composure       marking  standing_tackle  sliding_tackle  \
count  17954.000000  17954.000000     17954.000000    17954.000000   
mean      58.680183     47.162861        47.733040       45.705915   
std       11.625541     20.037346        21.674973       21.285812   
min       12.000000      3.000000         2.000000        3.000000   
25%       51.000000     30.000000        27.000000       24.000000   
50%       60.000000     52.500000        55.000000       52.000000   
75%       67.000000     64.000000        66.000000       64.000000   
max       96.000000     94.000000        93.000000       90.000000   

          gk_diving   gk_handling    gk_kicking  gk_positioning   gk_reflexes  
count  17954.000000  17954.000000  17954.000000    17954.000000  17954.000000  
mean      16.784449     16.556199     16.388660       16.541049     16.874067  
std       17.893895     17.108234     16.701447       17.231495     18.182943  
min        1.000000      1.000000      1.000000        1.000000      1.000000  
25%        8.000000      8.000000      8.000000        8.000000      8.000000  
50%       11.000000     11.000000     11.000000       11.000000     11.000000  
75%       14.000000     14.000000     14.000000       14.000000     14.000000  
max       90.000000     92.000000     92.000000       90.000000     94.000000  

[8 rows x 50 columns]

### Data Description
The dataframe has 17953 rows and 92 columns that has Fifa players data

In [5]:
#  Get the data types of each column.
df.dtypes

id             int64
name          object
full_name     object
birth_date    object
age            int64
               ...  
lb            object
lcb           object
cb            object
rcb           object
rb            object
Length: 92, dtype: object

In [6]:
#  Get a list of all columns within the DataFrame.
df.columns

Index(['id', 'name', 'full_name', 'birth_date', 'age', 'height_cm',
       'weight_kgs', 'positions', 'nationality', 'overall_rating', 'potential',
       'value_euro', 'wage_euro', 'preferred_foot',
       'international_reputation_1_5', 'weak_foot_1_5', 'skill_moves_1_5',
       'work_rate', 'body_type', 'release_clause_euro', 'club_team',
       'club_rating', 'club_position', 'club_jersey_number', 'club_join_date',
       'contract_end_year', 'national_team', 'national_rating',
       'national_team_position', 'national_jersey_number', 'crossing',
       'finishing', 'heading_accuracy', 'short_passing', 'volleys',
       'dribbling', 'curve', 'freekick_accuracy', 'long_passing',
       'ball_control', 'acceleration', 'sprint_speed', 'agility', 'reactions',
       'balance', 'shot_power', 'jumping', 'stamina', 'strength', 'long_shots',
       'aggression', 'interceptions', 'positioning', 'vision', 'penalties',
       'composure', 'marking', 'standing_tackle', 'sliding_tackle',
     

In [7]:
# Rename the headers to be more explanatory
df.rename(columns=str.title, inplace =True)
df

Id            Name                         Full_Name  Birth_Date  \
0      158023        L. Messi    Lionel Andrés Messi Cuccittini  1987-06-24   
1      190460      C. Eriksen      Christian  Dannemann Eriksen  1992-02-14   
2      195864        P. Pogba                        Paul Pogba  1993-03-15   
3      198219      L. Insigne                   Lorenzo Insigne  1991-06-04   
4      201024    K. Koulibaly                 Kalidou Koulibaly  1991-06-20   
...       ...             ...                               ...         ...   
17949  204322     R. McKenzie                    Rory  McKenzie  1993-10-07   
17950  239762       M. Sipľak                     Michal Sipľak  1996-02-02   
17951  235155      J. Bekkema                       Jan Bekkema  1996-04-09   
17952  244883      A. Al Yami               Abdulrahman Al Yami  1997-06-19   
17953  247187  Júnior Brumado  José Francisco dos Santos Júnior  1999-05-15   

       Age  Height_Cm  Weight_Kgs  Positions   Nationality  Overall_Rating  \
0       31     170.18        72.1   CF,RW,ST     Argentina              94   
1       27     154.94        76.2  CAM,RM,CM       Denmark              88   
2       25     190.50        83.9     CM,CAM        France              88   
3       27     162.56        59.0      LW,ST         Italy              88   
4       27     187.96        88.9         CB       Senegal              88   
...    ...        ...         ...        ...           ...             ...   
17949   25     175.26        74.8  RM,CAM,CM      Scotland              67   
17950   23     182.88        79.8         LB      Slovakia              59   
17951   22     185.42        89.8         GK   Netherlands              59   
17952   21     175.26        64.9      ST,LM  Saudi Arabia              59   
17953   19     190.50        79.8         ST        Brazil              59   

       ...   Lwb   Ldm   Cdm   Rdm   Rwb    Lb   Lcb    Cb   Rcb    Rb  
0      ...  64+2  61+2  61+2  61+2  64+2  59+2  48+2  48+2  48+2  59+2  
1      ...  71+3  71+3  71+3  71+3  71+3  66+3  57+3  57+3  57+3  66+3  
2      ...  76+3  77+3  77+3  77+3  76+3  74+3  72+3  72+3  72+3  74+3  
3      ...  63+3  58+3  58+3  58+3  63+3  58+3  44+3  44+3  44+3  58+3  
4      ...  73+3  77+3  77+3  77+3  73+3  76+3  85+3  85+3  85+3  76+3  
...    ...   ...   ...   ...   ...   ...   ...   ...   ...   ...   ...  
17949  ...  53+2  52+2  52+2  52+2  53+2  50+2  46+2  46+2  46+2  50+2  
17950  ...  57+2  55+2  55+2  55+2  57+2  57+2  58+2  58+2  58+2  57+2  
17951  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
17952  ...  41+2  35+2  35+2  35+2  41+2  39+2  32+2  32+2  32+2  39+2  
17953  ...  41+2  40+2  40+2  40+2  41+2  40+2  40+2  40+2  40+2  40+2  

[17954 rows x 92 columns]

In [8]:
#  Get a list of all columns within the DataFrame.
df.columns

Index(['Id', 'Name', 'Full_Name', 'Birth_Date', 'Age', 'Height_Cm',
       'Weight_Kgs', 'Positions', 'Nationality', 'Overall_Rating', 'Potential',
       'Value_Euro', 'Wage_Euro', 'Preferred_Foot',
       'International_Reputation_1_5', 'Weak_Foot_1_5', 'Skill_Moves_1_5',
       'Work_Rate', 'Body_Type', 'Release_Clause_Euro', 'Club_Team',
       'Club_Rating', 'Club_Position', 'Club_Jersey_Number', 'Club_Join_Date',
       'Contract_End_Year', 'National_Team', 'National_Rating',
       'National_Team_Position', 'National_Jersey_Number', 'Crossing',
       'Finishing', 'Heading_Accuracy', 'Short_Passing', 'Volleys',
       'Dribbling', 'Curve', 'Freekick_Accuracy', 'Long_Passing',
       'Ball_Control', 'Acceleration', 'Sprint_Speed', 'Agility', 'Reactions',
       'Balance', 'Shot_Power', 'Jumping', 'Stamina', 'Strength', 'Long_Shots',
       'Aggression', 'Interceptions', 'Positioning', 'Vision', 'Penalties',
       'Composure', 'Marking', 'Standing_Tackle', 'Sliding_Tackle',
     

In [9]:
df.nunique()

Id            17954
Name          16995
Full_Name     17898
Birth_Date     6156
Age              29
              ...  
Lb               97
Lcb             109
Cb              109
Rcb             109
Rb               97
Length: 92, dtype: int64

In [10]:
df['Positions'].value_counts()

CB              2243
GK              2065
ST              1747
CM               764
CDM,CM           709
                ... 
LB,LWB,CM          1
CAM,CF,CM,RM       1
RW,RM,ST,LW        1
RM,CF,LM,CAM       1
LM,CAM,LB,LW       1
Name: Positions, Length: 890, dtype: int64

In [11]:
df['Club_Position'].value_counts()

SUB    7870
RES    2885
LCB     655
RCB     654
GK      644
LB      549
RB      547
ST      449
RM      417
LM      414
RCM     413
LCM     411
CAM     308
LDM     235
RDM     234
RS      205
LS      204
CDM     147
LW      144
RW      144
CB      114
CM       80
RWB      67
LWB      66
RAM      21
LAM      20
RF       16
LF       16
CF       11
Name: Club_Position, dtype: int64

In [12]:
df[df.Positions == 'GK']

Id           Name              Full_Name  Birth_Date  Age  \
8      167495       M. Neuer           Manuel Neuer  1986-03-27   32   
11     192119    T. Courtois       Thibaut Courtois  1992-05-11   26   
12     192448  M. ter Stegen  Marc-André ter Stegen  1992-04-30   26   
27     193041       K. Navas           Keylor Navas  1986-12-15   32   
32     206585           Kepa      Kepa Arrizabalaga  1994-10-03   24   
...       ...            ...                    ...         ...  ...   
17921  162835  S. Handanovič       Samir Handanovič  1984-07-14   34   
17923  167948      H. Lloris            Hugo Lloris  1986-12-26   32   
17936  200389       J. Oblak              Jan Oblak  1993-01-07   26   
17942  193080         De Gea  David De Gea Quintana  1990-11-07   28   
17951  235155     J. Bekkema            Jan Bekkema  1996-04-09   22   

       Height_Cm  Weight_Kgs Positions  Nationality  Overall_Rating  ...  Lwb  \
8         193.04        92.1        GK      Germany              89  ...  NaN   
11        198.12        96.2        GK      Belgium              89  ...  NaN   
12        187.96        84.8        GK      Germany              89  ...  NaN   
27        185.42        79.8        GK   Costa Rica              87  ...  NaN   
32        185.42        84.8        GK        Spain              84  ...  NaN   
...          ...         ...       ...          ...             ...  ...  ...   
17921     193.04        92.1        GK     Slovenia              88  ...  NaN   
17923     187.96        82.1        GK       France              88  ...  NaN   
17936     187.96        87.1        GK     Slovenia              90  ...  NaN   
17942     193.04        76.2        GK        Spain              91  ...  NaN   
17951     185.42        89.8        GK  Netherlands              59  ...  NaN   

       Ldm  Cdm  Rdm  Rwb   Lb  Lcb   Cb  Rcb   Rb  
8      NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
11     NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
12     NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
27     NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
32     NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
...    ...  ...  ...  ...  ...  ...  ...  ...  ...  
17921  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
17923  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
17936  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
17942  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
17951  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[2065 rows x 92 columns]

In [13]:
split_positions = df["Positions"].str.split(',',n=3,expand=True).rename(columns={0:"A",1:"B",2:"C",3:"D"})

In [14]:
split_positions['A'].value_counts()

CB     3084
ST     2516
CM     2155
GK     2065
CDM    1386
RB     1342
LB     1324
RM     1052
CAM    1046
LM     1029
LW      378
RW      356
CF       88
RWB      68
LWB      65
Name: A, dtype: int64

In [15]:
df['Body_Type'].value_counts()

Normal                 10393
Lean                    6468
Stocky                  1086
PLAYER_BODY_TYPE_25        1
Akinfenwa                  1
Messi                      1
Shaqiri                    1
Neymar                     1
Courtois                   1
C. Ronaldo                 1
Name: Body_Type, dtype: int64

In [16]:
df['Club_Team'].value_counts()

Eintracht Frankfurt    33
Southampton            33
FC Nantes              33
FC Augsburg            33
Fortuna Düsseldorf     33
                       ..
Greece                  2
Wales                   2
Republic of Ireland     1
Poland                  1
Italy                   1
Name: Club_Team, Length: 687, dtype: int64

In [17]:
df['National_Team'].value_counts()

United States               23
Scotland                    23
Denmark                     23
England                     23
Germany                     23
                            ..
Club Tijuana                 1
Neuchâtel Xamax              1
Urawa Red Diamonds           1
Querétaro                    1
FC Admira Wacker Mödling     1
Name: National_Team, Length: 103, dtype: int64

In [18]:
df['Nationality'].value_counts()

England             1658
Germany             1199
Spain               1070
France               925
Argentina            904
                    ... 
Papua New Guinea       1
New Caledonia          1
St Lucia               1
Kuwait                 1
Nicaragua              1
Name: Nationality, Length: 160, dtype: int64

In [202]:
df = pd.DataFrame(df, columns=['Positions'])
df['Positions'] = df['Positions'].replace(['CF','RF','LF','ST'],'FW')
df['Positions'] = df['Positions'].replace(['AM','CAM','CDM','LM','RM','RW','LW','DM'],'MF')
df['Positions'] = df['Positions'].replace(['SW','RWB','LWB','RB','LB','RB','CB','LH','LCB','WB'],'DF')

df['Positions'] = df['Positions'].replace(['CB,CDM,CM,CAM',
                                           'LB,LWB,CB,LM','LB,LM,CB','LB,LM,CDM','LB,LW,CM','LB,LW,CM','LB,LWB,CM',
                                           'LB,RB,RM,LM','LB,CM,LWB',
                                           'LWB,LM,LB,CM','LW,CAM,CM,RB',
                                           'RB,RWB,RM,LWB','RB,CDM,RM,LB','RB,CB,CM,CDM','RB,RW,CM','RB,RW,RM',
                                           'RB,LB,LWB,RWB','RWB,LWB,LB,RB','RB,RM,LM,CM','RB,CDM,CB,CM','RB,RM,RW',
                                           'RB,CDM,CB,CM','RB,LW,LB,RW','RB,RM,LM,CM','RB,CAM','RB,CB,LB,CDM',
                                           'RWB,LM,ST','RWB,RB,LWB','RWB,LM,LWB,RB'],'DF')



df['Positions'] = df['Positions'].replace(['CAM,RM,CM','CAM,CDM,CM,RM','CAM,CF,CM,RM','CAM,LM,CM,CF','CAM,ST,CF,LM','CAM,CDM',
                                           'CAM,CDM,CM','CAM,CF','CAM,CDM,RM,CM','CAM,CDM,RM','CAM,CF,RW','CAM,CF,LM',
                                           'CAM,CF,CM','CAM,CF,ST','CAM,CF,RW,RM','CAM,CF,RM','CAM,CM','CAM,CF,ST,RM',
                                           'CAM,CM,CDM','CAM,CF,ST,RM','CAM,CM,CF','CAM,CM,CDM,RM','CAM,CM,LM','CAM,CM,CF,RM',
                                           'CAM,CM,CF,CDM','CAM,CM,LM,RM','CAM,CM,LM,CDM','CAM,CM,LW','CAM,CM,LM,ST',
                                           'CAM,CM,RM','CAM,CM,RM,LM','CAM,CM,RM,CF','CAM,CM,RW','CAM,CM,RM,ST','CAM,CM,RM,RW',
                                           'CAM,CM,ST','CAM,CM,RWB','CAM,CM,RW,RM','CAM,CM,RW,LW','CAM,RM,RW,CF',
                                           'CAM,LM','CAM,LB,CM,RM','CAM,LB,CM','CAM,LB','CAM,LM,CM',
                                           'CAM,LM,CF','CAM,LM,CDM','CAM,LB','CAM,LM,LW','CAM,LM,LB','CAM,LM,CM,ST',
                                           'CAM,LM,ST','CAM,LM,RW','CAM,LM,RM,ST','CAM,LW,CM',
                                           'CAM,RM,LM','CAM,RM,CM,ST','CAM,RM,CM,LM'
                                           'CAM,LM,RM','CAM,LM,LWB,LB','CAM,LM,LW,CM','CAM,LM,RM,RW','CAM,LM,RM,LB','CAM,LM,RM,CM',
                                           'CAM,LM,RM,CF','CAM,ST,CF,CM','CAM,LW','CAM,LM,ST,RM','CAM,LM,ST,CM',
                                           'CAM,LW,RW','CAM,LW,RB','CAM,LW,RB','CAM,LW,LM,ST','CAM,RM','CAM,RB',
                                           'CAM,LW,ST','CAM,RM,CF','CAM,RM,CDM',
                                           'CDM,CM,LM,CAM','CDM,CM,RB,LM','CDM,RM,CM','CDM,RWB','CDM,RM,CM','CDM,RM,CM',
                                           'CDM,RM,CM,CAM','CDM,CM,CB,LM','CAM,LW,LM,CM','CAM,LW,CM,RM','CAM,LW,CF',
                                           'CM,RW,LM','CM,CAM,RM,CDM','CM,CAM','CM,CF,RB','CM,CDM,LW','CM,RM,CF',
                                           'CM,LM,RM,CAM','CM,CDM,CAM,CF','CM,CAM,LM,RM',
                                           'LM,CAM,LB,LW','LM,RM,CM,ST','LM,CM,LW','LM,CM,LWB','LM,LWB,RM,RWB',
                                           'LM,CM,CDM,RM','LM,LW,CM','LM,RM,RWB','LM,LW,RM,ST','LM,CM,RM,CDM',
                                           'LM,ST,RW','LM,RM,ST,LB','LW,LB,LM','LM,CM,RM,RB',
                                           'LW,LM,CF','LW,LM,CF','LW,ST,LM,CAM','LW,RW,CF,ST','LW,ST','LW,RM,RW',
                                           'LW,RW,CAM,LM',
                                           'RM,CAM,CM','RM,RWB,ST','RM,LM,RWB','RM,LM,RW,LWB','RM,RW,CF',
                                           'RW,LW,CF','RW,CAM,CM','RW,CAM,CM','RM,ST,LM,CAM','RW,ST,RM,LW',
                                           'RM,RW,CM,CAM','RW,RM,LW,LM','RW,ST,CM',
                                           'RWB,RM,RB,LWB','RWB,RM,RB,LWB','RWB,RB,RM,CM','RWB,RM,RB,LB',
                                           'RWB,RM,RB','RWB,RM,LM','RWB,RM',
                                          ],'MF')



df['Positions'] = df['Positions'].replace(['CF,RW,ST','CF,ST,LM,RM','CF,ST,CAM,LM','CF,LW,ST'
                                           'ST,LM','ST,LW','ST,CF','ST,RB','ST,RW','ST,LM,CF','ST,CB',
                                           'ST,RM,RW,LM','ST,RWB,RM','ST,RW,RM','ST,RW,LW,CF','ST,RW,LW,CAM',
                                           'ST,RW,LW','ST,RW,LM,LW','ST,RW,LM','ST,RW,CF,RM','ST,RW,CF','ST,RW,CAM',
                                           'ST,RM,RWB,LM','ST,RM,RW,LW','ST,RM,RW','ST,RM,LW','ST,RM,LM,LW',
                                           'ST,RM,LM,CF','ST,RM,LM,CAM','ST,RM,LM','ST,RM,CAM,LM','ST,RM,CAM','ST,RM',
                                           'ST,LW,RW,LM','ST,LW,RW,CAM','ST,LW,RW','ST,LW,RW','ST,LW,RM','ST,LW,LM,CF',
                                           'ST,LW,LM','ST,LW,CM','ST,LW,CDM','ST,LW,CAM','ST,CF,RW',
                                           'ST,LM,LWB','ST,LM,LW,RW','ST,LM,LW','ST,LM,CAM','ST,CM,RB','ST,CM',
                                           'ST,CAM,RW,LW','ST,CAM,RW,LW','ST,CAM,RW','ST,CAM,RM,LM','ST,CAM,RM','ST,CAM,LW,RW',
                                           'ST,CAM,LM,CM','ST,CAM,LM,CF','ST,CAM,LM','ST,CAM,CF',
                                           'ST,CAM,RM,LM','ST,CAM,RM','ST,CAM,LW,RW','ST,CAM,LM,CM','ST,CAM',
                                           'ST,CF,RM','ST,CF,LW','ST,CF,LM,RW','ST,CF,LM,CAM','ST,CF,CAM,LM','ST,CF,CAM',
                                           'ST,LM,RW,RM','ST,LM,RW','ST,LM,RM,CM','ST,LM,RM,CF','ST,LM,RM','ST,LM,RM,CF'
                                          ],'FW')
df
3151	'RWB,RM,RB,LB','RWB,RM,RB','RWB,RM,LM','RWB,RM'
...	...
17296	

Positions
4649         FW
4074         FW
13876        FW
4878         FW
9084         FW
...         ...
14887        MF
17715        MF
16885        MF
15162        MF
9754         MF

[17954 rows x 1 columns]

In [203]:
df['Positions'].value_counts()

DF               3562
FW               2571
MF               2100
GK               2065
CM                764
                 ... 
LW,LM,RM,RW         1
RM,CM,LM,RB         1
CF,LW,ST            1
CF,LM               1
LB,LWB,CAM,LM       1
Name: Positions, Length: 661, dtype: int64

In [204]:
df['Positions'].nunique()

661

In [205]:
df = df.sort_values('Positions',ascending=False)
df

Positions
3151   RWB,RM,RB,LB
6957      RWB,RM,RB
3417      RWB,RM,RB
10728     RWB,RM,LM
5248         RWB,RM
...             ...
17296     CAM,RM,LM
13946     CAM,RM,LM
10551  CAM,RM,CM,ST
7911   CAM,RM,CM,ST
10748  CAM,RM,CM,LM

[17954 rows x 1 columns]

In [210]:
position_df= df['Positions'].to_list()
print('position_df: ', position_df)


position_df:  ['RWB,RM,RB,LB', 'RWB,RM,RB', 'RWB,RM,RB', 'RWB,RM,LM', 'RWB,RM', 'RWB,RM', 'RWB,RM', 'RWB,RM', 'RWB,RM', 'RWB,RM', 'RWB,RM', 'RWB,RB,RW', 'RWB,RB,RM', 'RWB,RB,RM', 'RWB,RB,RM', 'RWB,RB,RM', 'RWB,RB,RM', 'RWB,RB,RM', 'RWB,RB,RM', 'RWB,RB,RM', 'RWB,RB,CDM,RM', 'RWB,RB,CDM', 'RWB,RB,CDM', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,RB', 'RWB,LWB', 'RWB,LWB', 'RWB,LW,RW', 'RWB,LB', 'RWB,LB', 'RWB,CB', 'RW,ST,RM', 'RW,ST,RM', 'RW,ST,RM', 'RW,ST,RM', 'RW,ST,LW,CF', 'RW,ST,LW', 'RW,ST,LW', 'RW,ST,LW', 'RW,ST,LW', 'RW,ST,CAM,RM', 'RW,ST,CAM', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,ST', 'RW,

In [216]:
# initializing list
test_list = position_df

res = []
for i in test_list:
    if i not in res:
        res.append(i)
# printing list after removal 
print (res)

['RWB,RM,RB,LB', 'RWB,RM,RB', 'RWB,RM,LM', 'RWB,RM', 'RWB,RB,RW', 'RWB,RB,RM', 'RWB,RB,CDM,RM', 'RWB,RB,CDM', 'RWB,RB', 'RWB,LWB', 'RWB,LW,RW', 'RWB,LB', 'RWB,CB', 'RW,ST,RM', 'RW,ST,LW,CF', 'RW,ST,LW', 'RW,ST,CAM,RM', 'RW,ST,CAM', 'RW,ST', 'RW,RWB', 'RW,RM,ST,LW', 'RW,RM,ST', 'RW,RM,LW,ST', 'RW,RM,LM,ST', 'RW,RM,LM', 'RW,RM,LB,CM', 'RW,RM,CM,CAM', 'RW,RM,CM', 'RW,RM,CF,CAM', 'RW,RM,CF', 'RW,RM,CAM,ST', 'RW,RM,CAM', 'RW,RM', 'RW,RB', 'RW,LW,ST', 'RW,LW,RM,ST', 'RW,LW,RM', 'RW,LW,LM', 'RW,LW,CM', 'RW,LW,CAM,ST', 'RW,LW,CAM,CM', 'RW,LW,CAM', 'RW,LW', 'RW,LM,RM,ST', 'RW,LM,RM', 'RW,LM,CAM', 'RW,LM', 'RW,LB', 'RW,CM,RM', 'RW,CM,CAM,LW', 'RW,CM,CAM', 'RW,CM', 'RW,CF,LW', 'RW,CF', 'RW,CAM,ST,LW', 'RW,CAM,ST', 'RW,CAM,RM,CF', 'RW,CAM,RM', 'RW,CAM,LW', 'RW,CAM,CF', 'RW,CAM', 'RM,ST,RW', 'RM,ST,LM,RW', 'RM,ST,LM,LW', 'RM,ST,LM', 'RM,ST,CAM', 'RM,ST', 'RM,RWB,RW,CM', 'RM,RWB,RB', 'RM,RWB,LM,CAM', 'RM,RWB,LM', 'RM,RWB', 'RM,RW,ST,LM', 'RM,RW,ST,CAM', 'RM,RW,ST', 'RM,RW,RWB', 'RM,RW,RB,RWB', 'RM,R

In [218]:
res(ascending=False)

TypeError: 'list' object is not callable